File: fibonacci.py
#!/usr/bin/env python3

def fibonacci_generator():
    """Generator that yields fibonacci numbers indefinitely."""
    a, b = 0, 1
    while True:
        yield a
        a, b = b, a + b

def print_first_n_fibonacci(n):
    """Print the first n fibonacci numbers."""
    fib_gen = fibonacci_generator()
    for i in range(n):
        print(f"{i + 1}: {next(fib_gen)}")

if __name__ == "__main__":
    print("First 1000 Fibonacci numbers:")
    print_first_n_fibonacci(1000)

File: fibonacci.ts
function generateFibonacci(count: number): bigint[] {
    const fibonacci: bigint[] = [];
    
    if (count <= 0) return fibonacci;
    if (count >= 1) fibonacci.push(0n);
    if (count >= 2) fibonacci.push(1n);
    
    for (let i = 2; i < count; i++) {
        fibonacci.push(fibonacci[i - 1] + fibonacci[i - 2]);
    }
    
    return fibonacci;
}

function printFirstNFibonacci(n: number): void {
    const fibNumbers = generateFibonacci(n);
    
    console.log(`First ${n} Fibonacci numbers:`);
    fibNumbers.forEach((num, index) => {
        console.log(`${index + 1}: ${num}`);
    });
}

if (require.main === module) {
    printFirstNFibonacci(1000);
}

export { generateFibonacci, printFirstNFibonacci };